### ISSUE TITLE
Speed up CI

### ISSUE THREAD

**Issue by ChakshuGautam** [Samagra-Development/WarpSQL#36]
Currently, the CI takes up ~20ish minutes which is too slow for my liking.

**Comment by PiyushRaj927** [comment]
We should try using https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows to speed up zombodb's build time.

**Comment by ChakshuGautam** [comment]
I believe it would be an iterative process with a lot of hacks. Let's start with what we know right now.

Let me know if you want to start working on this.

**Comment by PiyushRaj927** [comment]
Sure, I'd be happy to start working on this.

**Comment by PiyushRaj927** [comment]
@ChakshuGautam, I have reduced the build time of WarpSQL to under 2 minutes when rebuilding the same image by implementing Docker layer caching. The improvement can be seen here: https://github.com/PiyushRaj927/WarpSQL/actions/runs/5165400680/jobs/9304900040#logs. I will create a pull request soon.

Regarding the ZomboDB cache, I realized that my initial idea would be challenging to implement. Utilizing Docker's built-in layer caching to cache the ZomboDB build is a more effective solution.
